src/snapshots.rs

```diff
@@ -78,23 +82,26 @@ pub fn recover_snapshots(
 }
 
 pub fn recover_full_snapshot(
+    temp_dir: Option<&str>,
     snapshot_path: &str,
     storage_dir: &str,
     force: bool,
     this_peer_id: PeerId,
     is_distributed: bool,
 ) -> Vec<String> {
-    let temporary_dir = Path::new(storage_dir).join("snapshots_recovery_tmp");
-    std::fs::create_dir_all(&temporary_dir).unwrap();
+    let snapshot_temp_path = temp_dir.map(PathBuf::from).unwrap_or_else(|| {
+        Path::new(storage_dir).join(format!("snapshots-recovery-{}", Uuid::new_v4()))
```

Comment:
Why do we need a uuid here?

Reply:
Indeed not needed because it is called once. Reverted it to how it was before. Thanks!